### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by zuriguardiola**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,117 @@
-# Ministry of Justice Template Repository
+# Modernisation Platform Terraform Loadbalancer Module with Access Logs enabled
+[![repo standards badge](https://img.shields.io/badge/dynamic/json?color=blue&style=for-the-badge&logo=github&label=MoJ%20Compliant&query=%24.data%5B%3F%28%40.name%20%3D%3D%20%22modernisation-platform-terraform-s3-bucket%22%29%5D.status&url=https%3A%2F%2Foperations-engineering-reports.cloud-platform.service.justice.gov.uk%2Fgithub_repositories)](https://operations-engineering-reports.cloud-platform.service.justice.gov.uk/github_repositories#modernisation-platform-terraform-s3-bucket "Link to report")
```
Comment (davidkelliott) [inline README.md:2]: The 2 references to `modernisation-platform-terraform-s3-bucket` should be updated to `modernisation-platform-terraform-loadbalancer`

File: README.md [hunk 2]
```diff
@@ -1,16 +1,117 @@
...
+```
 
-* Edit the copy of this README.md file to document your project
-* Grant permissions to the appropriate MoJ teams
-* Setup branch protection
-* Add a Dependabot file for [automated pull requests for package updates](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/enabling-and-disabling-dependabot-version-updates#enabling-dependabot-version-updates)
+Loadbalancer target groups and listeners need to be created separately.
 
-[create a repository]: https://github.com/ministryofjustice/template-repository/generate
+To run queries in Athen do the following:
```
Comment (davidkelliott) [inline README.md]:
```suggestion
To run queries in Athena do the following:
```

File: main.tf [hunk 3]
```diff
@@ -0,0 +1,213 @@
+data "aws_vpc" "shared" {
+  tags = {
+    "Name" = var.vpc_all
+  }
+}
+
+data "aws_kms_key" "s3" {
```
Comment (davidkelliott) [inline main.tf:7]: Should this be hard coded?

Comment (dms1981) [inline main.tf:7]: Should this key be hard-coded? I'd have thought a data call would be a better way of finding the correct key here.

File: main.tf [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+  key_id = "arn:aws:kms:eu-west-2:938841708445:key/77b5bfe6-f106-4052-9d26-a5bfefd2dff4"
+}
+
+# data "aws_subnets" "shared-public" {
```
Comment (davidkelliott) [inline main.tf:11]: Do we need this commented out data block?

**Review by davidkelliott — CHANGES_REQUESTED** [review]
Looking good!  I'd love to see a demo of this working! Few minor comments, also I think you'll need an output for the loadbalancer so people can get things like the ARN of the loadbalancer etc.

File: main.tf [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+#     values = [data.aws_vpc.shared.id]
+#   }
+#   tags = {
+#     Name = "${var.public_subnets}-public*"
+#   }
+# }
+
+# Terraform module which creates S3 Bucket resources for Load Balancer Access Logs on AWS.
+
+module "s3-bucket" {
```
Comment (dms1981) [inline main.tf:23]: How do we cater for customers who have a pre-existing log bucket, or don't want to define a separate bucket just for LB logs?

File: main.tf [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+  tags = merge(
+    var.tags,
+    {
+      Name = "lb-${var.application_name}"
+    },
+  )
+}
+
+resource "aws_security_group" "lb" {
```
Comment (dms1981) [inline main.tf:146]: What about egress rules?

File: main.tf [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+# https://www.terraform.io/docs/providers/aws/d/region.html
+# Get the region of the callee
+data "aws_region" "current" {}
+
+resource "aws_lb" "loadbalancer" {
+  name               = "${var.application_name}-lb"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.lb.id]
+  subnets            = [var.public_subnets[0], var.public_subnets[1], var.public_subnets[2]]
```
Comment (dms1981) [inline main.tf:128]: Can this just be a list supplied when the module is called?

Comment (zuriguardiola) [reply main.tf:128]: 🤔 how do you mean?

File: main.tf (hunk 6 above)
Comment (zuriguardiola) [reply main.tf:147]: Added egress options

**Review by davidkelliott — APPROVED** [review]
Looks good thanks Zuri 👍

**Pull request merged**